DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateProgressController.cs

```diff
@@ -84,12 +85,58 @@ DelegateProgressAccessRoute accessedVia
 
             progressService.UpdateSupervisor(progressId, formData.SupervisorId);
 
+            return ReturnToPreviousPage(formData.DelegateId, progressId, accessedVia);
+        }
+
+        [HttpGet]
+        [Route("EditCompleteByDate")]
+        public IActionResult EditCompleteByDate(int progressId, DelegateProgressAccessRoute accessedVia)
+        {
+            var centreId = User.GetCentreId();
+            var delegateCourseProgress =
+                courseService.GetDelegateCourseProgress(progressId, centreId);
+
+            var model = new EditCompleteByDateViewModel(
+                progressId,
+                accessedVia,
+                delegateCourseProgress!.DelegateCourseInfo
+            );
+            return View(model);
+        }
+
+        [HttpPost]
+        [Route("EditCompleteByDate")]
+        public IActionResult EditCompleteByDate(
+            EditCompleteByDateFormData formData,
+            int progressId,
+            DelegateProgressAccessRoute accessedVia
+        )
+        {
+            if (!ModelState.IsValid)
+            {
+                var model = new EditCompleteByDateViewModel(formData, progressId, accessedVia);

```

Comment:
When there is an error the course name disappears so it just shows 'Edit complete date for'

Reply:
Resolved this for this this ticket.